File: Algorithms/Implementation/DesignerPdfViewer.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Implementation -> Designer PDF Viewer
 *	Easy
 */
int main(){
    int alphabet[26];
    for(int h_i = 0; h_i < 26; h_i++){
       scanf("%d",&alphabet[h_i]);
    }
    char* word = (char *)malloc(512000 * sizeof(char));
    scanf("%s",word);

    int length = strlen(word);

    int maxHeight = 0;
    for(int i = 0; i < length; i++) {
    	int height = alphabet[ word[i] - 'a' ];
    	if(height > maxHeight) maxHeight = height;
    }

    int area = length * maxHeight;

    printf("%d", area);

    return 0;
}

File: Algorithms/Implementation/DesignerPdfViewer.java
package Implementation;

import java.util.Scanner;

/**
 *	Algorithms -> Implementations -> Designer PDF Viewer
 *	Easy
 */
public class DesignerPdfViewer {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		int[] alphabetHeights = new int[26];
		for(int i = 0; i < 26; i++) {
			alphabetHeights[i] = in.nextInt();
		}
		
		String word = in.next();
		
		int maxHeight = 0;
		
		for(char letter: word.toCharArray()) {
			int height = alphabetHeights[letter - 'a'];
			if(height > maxHeight) maxHeight = height;
		}
		
		int area = maxHeight * word.length();
		
		System.out.println(area);
		
		in.close();
	}

}
